app/packages/looker/src/overlays/segmentation.ts

```diff
@@ -130,29 +129,80 @@ export default class SegmentationOverlay<State extends BaseState>
       maskTargets = coloring.defaultMaskTargets;
     }
 
-    const color =
-      maskTargets && Object.keys(maskTargets).length === 1
-        ? getColor(
-            state.options.coloring.pool,
-            state.options.coloring.seed,
-            this.field
-          )
-        : coloring.targets[
-            Math.round(Math.abs(target)) % coloring.targets.length
-          ];
+    const target = this.getTarget(state);
 
-    return {
-      color,
-      label: {
-        ...this.label,
-        mask: {
-          shape: this.label.mask.data.shape ? this.label.mask.data.shape : null,
+    if (this.label.mask.data.channels > 2) {
+      const rgbSegmentationInfoWithoutColor: Omit<
+        PointInfo<SegmentationInfo>,
+        "color"
+      > = {
+        label: {
+          ...this.label,
+          mask: {
+            shape: this.label.mask.data.shape
+              ? this.label.mask.data.shape
+              : null,
+          },
         },
-      },
-      field: this.field,
-      target,
-      type: "Segmentation",
-    };
+        field: this.field,
+        target,
+        type: "Segmentation",
+      };
+
+      if (!isRgbMaskTargets(maskTargets)) {
+        // getting color here is computationally inefficient, return no color for this edge case
+        return rgbSegmentationInfoWithoutColor;
```

Comment:
Can you elaborate here? I'm not sure I understand

Reply:
This is for the edge case when the mask is multi-channel (RGB) but the mask targets is not RGB (or is missing). In this scenario, we render the RGB mask as is but the tooltip will behave slightly differently in two respects:
1. The ribbon color won't reflect the segmentation color, it'll be static in the entire canvas (a shade of white, I think).
2. We won't display any "pixel" info.

This comment is re: (1); while it's possible to get the color of the underlying mask, it involves parsing the RGB buffer. In the happy path case, the RGB buffer is reduced into a targets lookup cable and we don't have this problem.